identity/src/commonMain/kotlin/com/android/identity/nfc/ServiceParameterRecord.kt

```diff
@@ -66,17 +67,19 @@ data class ServiceParameterRecord(
 
             val p = record.payload
             require(p.size >= 1) { "Unexpected length of Service Parameter Record" }
-            val serviceNameLen = p[1].toInt().and(0xff)
+            val serviceNameLen = p.getUInt8(1).toInt()
             require(p.size == serviceNameLen + 7) { "Unexpected length of body in Service Parameter Record" }
 
-            return ServiceParameterRecord(
-                tnepVersion = p[0].toInt().and(0xff),
-                serviceNameUri = p.toByteArray().decodeToString(2, serviceNameLen + 2),
-                tnepCommunicationMode = p[2 + serviceNameLen].toInt().and(0xff),
-                wtInt = p[3 + serviceNameLen].toInt().and(0xff),
-                nWait = p[4 + serviceNameLen].toInt().and(0xff),
-                maxNdefSize = (p[5 + serviceNameLen].toInt().and(0xff)) * 0x100 + (p[6 + serviceNameLen].toInt().and(0xff))
-            )
+            return with (ByteDataReader(p)) {
+                ServiceParameterRecord(
+                    tnepVersion = getUInt8().toInt(), // TODO: b/393388370 - 1 byte, then skip 1 byte? Looks wrong.
```

Comment:
No, there's nothing wrong about this - the byte we're skipping is `serviceNameLen` which we read right about. Please remove the TODO.